apps/emqx_rule_engine/src/emqx_rule_engine_api.erl

```diff
@@ -405,9 +406,11 @@ param_path_id() ->
                                 },
                                 #{tag => ?TAG}
                             ),
-                            {400, #{code => 'BAD_REQUEST', message => ?ERR_BADARGS(Reason)}}
+                            ?BAD_REQUEST(?ERR_BADARGS(Reason))
                     end
-            end
+            end;
+        _BadId ->
+            ?BAD_REQUEST(<<"rule id must be a string">>)
```

Comment:
To be 100% clear, the issue is that a user can create a rule where id is literal string "null". Does this patch handle it?

Reply:
Hmm I managed to reproduce the issue by passing the JSON null value, not a string `"null"`.  :thinking: 

I'll add an assertion for the `"null"` case, but I _think_ it shouldn't be a problem, so long it's a string.

Reply:
Turns out it is a bug in hocon: https://github.com/emqx/hocon/pull/299